### PR TITLE
Warn when @-file selected for context is too big

### PR DESCRIPTION
**Pull request by pkukielka**
## Changes

* Warning for too big @-files was added
* One TODO removed - is it now possible to switch between addition of few @-files just by moving cursor between them
* Minor code cleanup

image

## Test plan

Manual QA tests only

### CODE REVIEW

File: src/main/java/com/sourcegraph/cody/PromptPanel.kt [hunk 1]
```diff
@@ -264,12 +251,41 @@ class PromptPanel(project: Project, private val chatSession: ChatSession) : JLay
...
+    }
   }
 }
 
 data class DisplayedContextFile(val contextItem: ContextItem) {
   override fun toString(): String {
-    return displayPath(contextItem)
+    val contextItemFile = contextItem as? ContextItemFile
+    val isTooLarge = contextItemFile?.title == "large-file" || contextItemFile?.isTooLarge == true
+    val warnIfNeeded = if (isTooLarge) "<i> - ⚠ File too large</i>" else ""
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/cody/PromptPanel.kt:287]: Can we bundle this text?

File: src/main/java/com/sourcegraph/cody/PromptPanel.kt [hunk 2]
```diff
@@ -264,12 +251,41 @@ class PromptPanel(project: Project, private val chatSession: ChatSession) : JLay
...
+      return contextItems.filter { f -> atExpressions.any { it.value == f.displayPath() } }
+    }
   }
 }
 
 data class DisplayedContextFile(val contextItem: ContextItem) {
   override fun toString(): String {
-    return displayPath(contextItem)
+    val contextItemFile = contextItem as? ContextItemFile
+    val isTooLarge = contextItemFile?.title == "large-file" || contextItemFile?.isTooLarge == true
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/cody/PromptPanel.kt:286]: Can we extract this title somewhere? maybe as a field in ContextItem.kt

File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextItem.kt [hunk 3]
```diff
@@ -44,7 +53,8 @@ data class ContextItemFile(
     val source: ContextFileSource? =
         null, // Oneof: embeddings, user, keyword, editor, filename, search, unified, selection,
     // terminal
-    val content: String? = null
+    val content: String? = null,
+    val isTooLarge: Boolean? = null
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextItem.kt:57]: NOTE: this field does not exist yet but I see it is there in the later cody versions

File: src/main/java/com/sourcegraph/cody/PromptPanel.kt (hunk 2 above)
Comment (mkondratek) [inline src/main/java/com/sourcegraph/cody/PromptPanel.kt:286]: this handling should work now and later 💪

File: src/main/java/com/sourcegraph/cody/PromptPanel.kt [hunk 4]
```diff
@@ -6,34 +6,28 @@ import com.intellij.openapi.actionSystem.KeyboardShortcut
...
 import com.sourcegraph.cody.vscode.CancellationToken
 import com.sourcegraph.common.CodyBundle
 import java.awt.Dimension
-import java.awt.event.ComponentAdapter
-import java.awt.event.ComponentEvent
-import java.awt.event.KeyEvent
-import java.awt.event.MouseAdapter
-import java.awt.event.MouseEvent
-import java.io.File
+import java.awt.event.*
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/cody/PromptPanel.kt:24]: nit: wildcard import - I guess your intellij settings change that automatically

**Review by mkondratek — APPROVED** [review]
Tested locally on 2023.3. Works, no errors. lgtm 👍

File: src/main/java/com/sourcegraph/cody/PromptPanel.kt (hunk 1 above)
Comment (pkukielka) [reply src/main/java/com/sourcegraph/cody/PromptPanel.kt:287]: I'm doing a followup to include a file ranges, I will fix there if you don't mind.

**Pull request merged**